__main__.py

```diff
@@ -3,81 +3,149 @@
     ) -> None:
         self._data = data
-        self.date = registered_at or dt.datetime.now(tz=dt.timezone.utc)
+        self.time = time or now()
 
-    def __getitem__(self, key: str) -> str:
+    def __getitem__(self, key: PEP) -> Status:
         return self._data[key]
 
-    def __xor__(self, other: State) -> dict[str, tuple[str, str]]:
-        return {k: (self[k], other[k]) for k in self._data if self[k] != other[k]}
+    def __xor__(self, other: State) -> dict[PEP, tuple[Status, Status]]:
+        # FIXME Handle PEPs that have been newly added, and are only on `other`
+        return {k: (self[k], other[k]) for k in self._data if self[k] is not other[k]}
+
+    @classmethod
+    def from_api(cls, data: dict[str, dict[str, str]], /) -> Self:
+        return cls(data={PEP(k): Status(v["status"]) for k, v in data.items()})
+
+    @classmethod
+    def _migrate(cls, old: dict[str, Any] | StateDict, /) -> StateDict:
+        out = StateDict(version=cls.version, data={}, time=now().isoformat())
+        old_version = old.get("version")
+        if old_version is None:
+            data = cast("dict[str, str]", old.copy())
+            out["data"] = data
+        else:
+            out = cast(StateDict, old)
+        return out
+
+    @classmethod
+    def from_dict(cls, state: StateDict | dict[str, Any], /) -> Self:
+        state_dict = cls._migrate(state)
+        data_raw = state_dict["data"]
+        data = {PEP(k): Status(v) for k, v in data_raw.items()}
+        time = dt.datetime.fromisoformat(state_dict["time"])
+        return cls(data=data, time=time)
+
+    def to_dict(self) -> StateDict:
+        return StateDict(
+            version=self.version,
+            data={str(k): v.value for k, v in self._data.items()},
+            time=self.time.isoformat(),
+        )
+
 
-    def dump(self) -> str:
-        return json.dumps(self._data, indent=2)
+def write_latest(state: State) -> None:
+    LATEST.write_text(json.dumps(state.to_dict(), indent=2))
 
 
 def get_last_state() -> State:
     if not LATEST.exists():
-        dprint("&4latest.json not found.", file=sys.stderr)
-        LATEST.write_text(get_current_state().dump())
-        dexit("&2latest.json created. Run the program again.")
+        dprint(f"&4{LATEST.name} not found.", file=sys.stderr)
+        write_latest(get_current_state())
+        dexit(f"&2{LATEST.name} created. Run the program again.")
     with LATEST.open() as f:
-        return State(
-            json.load(f),
-            dt.datetime.fromtimestamp(LATEST.stat().st_mtime, tz=dt.timezone.utc),
-        )
+        return State.from_dict(json.loads(f.read()))
 
 
 def get_current_state() -> State:
     try:
         req = requests.get(URL, timeout=15)
-    except (requests.ReadTimeout):
+        req.raise_for_status()
+    except requests.ReadTimeout:
         dexit("&4Connection timed out.")
-    if req.status_code != 200:
+    except HTTPError:
```

Comment:
I think this is a better way of handling HTTP exceptions - by using `raise_for_status`.